src/core/events.js

```diff
@@ -1,44 +1,44 @@
 import EventEmitter from 'eventemitter2'
 import store from '../store/store'
+import watch from 'redux-watch'
+import isEqual from 'deep-equal'
+import mapKeys from 'object-loops/map-keys'
 import * as selectors from '../store/selectors'
-import { createValuesHashToValueSelector } from '../store/selectors/utils'
-import { mapKeys } from '../utils/func.js'
-import { subcribeByWatching } from './utils'
 
 const events = new EventEmitter()
 
 //these methods have been bounded to their object, since they will be used
 //more than once and inside of other functions too
 const getState = () => store.getState()
-const getCaptures = ()=> selectors.captureSelector(getState())
+const getCaptures = ()=> selectors.confirmedCaptures(getState())
 
-const getCapturesCompatible = ()=> mapKeys(getCaptures(), key => key + 'Capture')
+const getCapturesCompatible = () => mapKeys(getCaptures(), key => key + 'Capture')
 
 const subscribe = store.subscribe.bind(store)
-//this function allows to subscribe to a selector and listen for when it changes
-const subcribeToStoreByWatching = subcribeByWatching(getState, subscribe)
-
 
-const emitIfCaptureValueTrue = (captureType, eventSufix) => captureValue => {
-  if (captureValue) events.emit(captureType+eventSufix, getCaptures()[captureType])
+const subscribeByWatching = (getState, subscribe) => (selector, changeCallback) => {
+  const watcher = watch(()=>selector(getState()), null, isEqual)
+  subscribe(watcher(changeCallback))
 }
 
-const subscribeToCaptureValueAndEmit = (captureHashValueSelector, eventSuffix) => captureType =>
-        subcribeToStoreByWatching(createValuesHashToValueSelector(captureHashValueSelector, captureType),
-                                  emitIfCaptureValueTrue(captureType, eventSuffix))
-
-const subcribeToConfirmedCapture =
-        subscribeToCaptureValueAndEmit(selectors.isThereAValidAndConfirmedCapture, 'Capture')
+//this function allows to subscribe to a selector and listen for when it changes
+const subscribeToStoreByWatching = subscribeByWatching(getState, subscribe)
 
-subcribeToConfirmedCapture('document')
-subcribeToConfirmedCapture('face')
+const captureCallback = captures => {
```

Comment:
I worry this makes the method less readable and harder to debug - I can't see or get hold of the object passed to the function so easily. 

In this case I have made the change because we use the entire object.